Code/Framework/AzToolsFramework/AzToolsFramework/PropertyTreeEditor/PropertyTreeEditor.cpp

```diff
@@ -392,6 +392,11 @@ namespace AzToolsFramework
         }
 
         AZ_Warning("PropertyTreeEditor", false, "SetProperty - value type cannot be converted to the property's type.");
+        auto valRepr = value.type().ToFixedString();
+        AZ_Warning("PropertyTreeEditor", false, "SetProperty - value type %s", valRepr.c_str());
+        auto propRepr = pteNode.m_nodePtr->GetClassMetadata()->m_typeId.ToFixedString();
+        AZ_Warning("PropertyTreeEditor", false, "SetProperty - property's type %s", propRepr.c_str());
+
```

Comment:
With the way it's written, you would need [[maybe_unused]] on valRepr and propRepr to avoid unused variable warnings on release builds. Maybe consider this instead:
```suggestion
        AZ_Warning("PropertyTreeEditor", false, "SetProperty - value type (%s) cannot be converted to the property's type (%s).", 
            value.type().ToFixedString().c_str(), 
            pteNode.m_nodePtr->GetClassMetadata()->m_typeId.ToFixedString().c_str());
```